apps/prairielearn/elements/pl-file-upload/pl-file-upload.py

```diff
@@ -37,48 +110,89 @@ def add_format_error(data: pl.QuestionData, error_string: str) -> None:
 
 def prepare(element_html: str, data: pl.QuestionData) -> None:
     element = lxml.html.fragment_fromstring(element_html)
-    required_attribs = ["file-names"]
-    optional_attribs = []
+
+    # Either "file-names" or "optional-file-names" is required, which is checked separately
+    required_attribs = []
+    optional_attribs = ["file-names", "optional-file-names", "allow-blank"]
     pl.check_attribs(element, required_attribs, optional_attribs)
+    if not pl.has_attrib(element, "file-names") and not pl.has_attrib(
+        element, "optional-file-names"
+    ):
+        raise Exception(
+            "One of required attributes file-names or optional-file-names missing"
+        )
+
+    if pl.has_attrib(element, "file-names") and pl.has_attrib(element, "allow-blank"):
+        raise Exception(
```

Comment:
```suggestion
        raise ValueError(
```
Nit: Generally better to raise a more specific exception.

Reply:
Changed.